### PR TITLE
Add automation app support

### LINKED ISSUE #1231: Update `aad-app-reg.sh` to create automation admin account and use in CI/CD bundle registration
The current CI/CD uses resource owner password credentials to authenticate against the TRE API. This causes an issue in tenants with MFA enabled. As an alternative, we will create an App Registration to allow the automation to use the [client credentials flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-daemon-overview) (preferred for this scenario).

The steps required are: 
- [ ] update the `aad-app-reg.sh` script to allow an additional App to be created (as TRE Admin)
- [ ] update the bundle registration in  CI/CD to use this App as the identity when registering bundles

### PR DESCRIPTION
**Pull request by stuartleeks**
Closes #1231

- [X] update `aad-app-reg.sh` to allow an additional App to be created (with `TREAdmin` role)
- [X] update the bundle registration called from CI/CD to use this App as the identity when registering bundles if configured
- [x] update docs for `aad-app-reg.sh` with new argument

As part of the `aad-app-reg.sh` changes, it was hard to find a meaningful single-letter switch, so this PR also includes adding long-forms of switches (and updates the doc to use the long-form for readability).

Whilst updating `publish_register_bundles.sh` the `eval` sections of the script have also been removed to simplify the script (and calling code)

Documenting the steps to add create the automation account manually will be done for #1232

### CODE REVIEW

File: templates/core/.env.sample [hunk 1]
```diff
@@ -11,3 +11,7 @@ AAD_TENANT_ID=__CHANGE_ME__
 API_CLIENT_ID=__CHANGE_ME__
 API_CLIENT_SECRET=__CHANGE_ME__
 SWAGGER_UI_CLIENT_ID=__CHANGE_ME__
+
+# Automation accounts (e.g. bundle registration)
+AUTOMATION_ADMIN_ACCOUNT_CLIENT_ID=__CHANGE_ME__
+AUTOMATION_ADMIN_ACCOUNT_CLIENT_SECRET=__CHANGE_ME__
```
Comment (martinpeck) [inline templates/core/.env.sample:17]: can you add a new line at the end of file please?

File: scripts/aad-app-reg.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # Setup Script
 set -euo pipefail
 
-function usage()
+function show_usage()
```
Comment (martinpeck) [inline scripts/aad-app-reg.sh:6]: minor: you've changed name to `show_usage`. This is a better name, but is now inconsistent with other scripts (e.g. publish_register_bundle.sh also in this PR).

Meh....feel free to ignore.

**Review by martinpeck — APPROVED** [review]

File: templates/core/.env.sample (hunk 1 above)
Comment (stuartleeks) [reply templates/core/.env.sample:17]: done 😄

File: scripts/aad-app-reg.sh [hunk 3]
```diff
@@ -516,14 +548,102 @@ JSON
...
+
+    # This tag ensures the app is listed in "Enterprise applications"
+    az ad sp update --id $automationSpId --set tags="['WindowsAzureActiveDirectoryIntegratedApp']"
+
   # Grant admin consent for the delegated workspace scopes
   if [[ $grantAdminConsent -eq 1 ]]; then
-      echo "Granting admin consent for ${appName} Swagger UI app (service principal ID ${swaggerSpId})"
-      az ad app permission grant --id $swaggerSpId --api $apiAppId --scope "user_impersonation"
+      echo "Granting admin consent for ${appName} API app (service principal ID ${spId}) - NOTE: Directory admin privileges required for this step"
+      echo "SP: $automationSpId"
```
Comment (stuartleeks) [inline scripts/aad-app-reg.sh:637]: Need to remove this!

Comment (stuartleeks) [reply scripts/aad-app-reg.sh]: done

**Review by ross-p-smith — APPROVED** [review]

**Pull request merged**